src/parsers/transactions.ts

```diff
@@ -79,22 +80,15 @@ export const parseTransactions = (
   const allL2Transactions = existingTransactions.filter(tx =>
     isL2Network(tx.network || '')
   );
-
   const data = appended
     ? transactionData
     : dataFromLastTxHash(transactionData, existingTransactions);
 
-  const parsedNewTransactions = flatten(
-    data.map(txn =>
-      parseTransaction(
-        txn,
-        accountAddress,
-        nativeCurrency,
-        purchaseTransactionHashes,
-        network
-      )
-    )
+  const newTransactionPromises = data.map(txn =>
+    parseTransaction(txn, nativeCurrency, purchaseTransactionHashes, network)
   );
+  const newTransactions = await Promise.all(newTransactionPromises);
```

Comment:
Promise.all has the behavior that it rejects immediately if any of the input promises reject - so we could be in a situation where one of the transactions throws an error (and parseTransaction doesn't catch all) and screws it up for all the other transactions